blockstack_client/keys.py

```diff
@@ -21,124 +21,102 @@
-    NAME_PREORDER, NAME_REGISTRATION, NAME_UPDATE, NAME_TRANSFER, NAMESPACE_PREORDER, NAME_IMPORT, \
-    USER_ZONEFILE_TTL, CONFIG_PATH
 
 log = get_logger()
 
-def is_multisig( privkey_info ):
+
+def is_multisig(privkey_info):
     """
     Does the given private key info represent
     a multisig bundle?
     """
-    if type(privkey_info) != dict:
+    if not isinstance(privkey_info, dict):
         return False
 
-    if 'private_keys' not in privkey_info.keys():
+    if 'private_keys' not in privkey_info:
         return False
 
-    if 'redeem_script' not in privkey_info.keys():
+    if 'redeem_script' not in privkey_info:
         return False
 
     return True
 
 
-def is_encrypted_multisig( privkey_info ):
+def is_encrypted_multisig(privkey_info):
     """
     Does a given encrypted private key info
     represent an encrypted multisig bundle?
     """
-    if type(privkey_info) != dict:
+    if not isinstance(privkey_info, dict):
         return False
 
-    if 'encrypted_private_keys' not in privkey_info.keys():
+    if 'encrypted_private_keys' not in privkey_info:
         return False
 
-    if 'encrypted_redeem_script' not in privkey_info.keys():
+    if 'encrypted_redeem_script' not in privkey_info:
         return False
 
     return True
 
 
-def is_singlesig( privkey_info ):
+def is_singlesig(privkey_info):
     """
     Does the given private key info represent
     a single signature bundle? (i.e. one private key)?
     """
-    if type(privkey_info) not in [str, unicode]:
+    if not isinstance(privkey_info, (str, unicode)):
         return False
 
     try:
         virtualchain.BitcoinPrivateKey(privkey_info)
         return True
     except:
-        return False
+        pass
 
+    return False
 
-def singlesig_privkey_to_string( privkey_info ):
+
+def singlesig_privkey_to_string(privkey_info):
     """
     Convert private key to string
     """
     return virtualchain.BitcoinPrivateKey(privkey_info).to_wif()
 
 
-def multisig_privkey_to_string( privkey_info ):
+def multisig_privkey_to_string(privkey_info):
     """
     Convert multisig keys to string
     """
-    return ",".join( [virtualchain.BitcoinPrivateKey(pk).to_wif() for pk in privkey_info['private_keys']] )
+    return ','.join([singlesig_privkey_to_string(pk) for pk in privkey_info['private_keys']])
 
 
-def privkey_to_string( privkey_info ):
+def privkey_to_string(privkey_info):
     """
     Convert private key to string
     Return None on invalid
     """
-    if is_singlesig( privkey_info ):
-        return singlesig_privkey_to_string( privkey_info )
+    if is_singlesig(privkey_info):
+        return singlesig_privkey_to_string(privkey_info)
 
-    elif is_multisig( privkey_info ):
-        return multisig_privkey_to_string( privkey_info )
+    if is_multisig(privkey_info):
+        return multisig_privkey_to_string(privkey_info)
 
-    else:
-        return None
+    # TODO: check if we should be returning '' here
```

Comment:
I prefer None personally.  The other methods that call this one use it to print out private keys, and returning None (to me) is a more straightforward way of saying "this is an invalid key."